src/ic-certified-assets/src/lib.rs

```diff
@@ -684,6 +684,50 @@ fn check_url_decode() {
     assert_eq!(url_decode("/%e6"), Ok("/æ".to_string()));
 }
 
+fn redirect_to_url(host: &str, url: &str) -> Option<String> {
+    if let Some(host) = host.split(':').next() {
+        let host = host.trim();
+        if let Some(base) = host.strip_suffix(".raw.ic0.app") {
```

Comment:
What about `raw.ic0.app` itself?

Reply:
An asset canister isn't going to be hosted from raw.ic0.app, that just redirects to the dashboard.

Reply:
While there is a plan to remove the ability to specify the canister id in the query string, at the present moment that still works.  Here is an example, run on mainnet:

```
$ curl -v 'https://raw.ic0.app?canisterId=nvosk-ayaaa-aaaai-aa7za-cai'
> GET /?canisterId=nvosk-ayaaa-aaaai-aa7za-cai HTTP/2
> Host: raw.ic0.app

!doctype html><html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width"/><title>mainnet</title><base href="/"/><link rel="icon" href="favicon.ico"/><link rel="stylesheet" href="main.css"/><script defer="defer" src="index.js"></script></head><body><main><img src="logo.png" alt="DFINITY logo"/><form action="#"><label for="name">Enter your name: &nbsp;</label> <input id="name" alt="Name"/> <button type="submit">Click Me!</button></form><section id="greeting"></section></main></body></html>*
```

Also, any canister can provide an `http_request` query method, not just asset canisters.
```
$ curl 'https://raw.ic0.app?canisterId=n4nzw-wqaaa-aaaai-aa7yq-cai' 
<html><body>host is raw.ic0.app</body></html>
```

Some of the code:
```
  public query func http_request(request: HttpRequest): async HttpResponse {
    let host = getHost(request.headers);
    {
      status_code = 200;
      headers = request.headers;
      body = Text.encodeUtf8("<html><body>host is " # host # "</body></html>");
      streaming_strategy = null;
    }
  };
```
```

Reply:
I had no idea, I thought that was just for the local replica. Good to know. But I was talking about canisters in general, not just asset canisters.